site/it/tutorials/customization/basics.ipynb

```diff
@@ -1,480 +1,480 @@
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "eBPw8e8vrsom"
+      },
+      "source": [
+        "Le differenze più ovvie tra gli array NumPy e i `tf.Tensor` sono:\n",
+        "\n",
+        "1. I tensori possono essere supportati da acceleratori di memoria (come GPU, TPU).\n",
+        "2. I tensori sono immutabili."
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "Dwi1tdW3JBw6"
+      },
+      "source": [
+        "### Compatibilità con NumPy \n",
+        "\n",
+        "La conversione tra oggetti `tf.Tensor` di TensorFlow e un `ndarray` di NumPy è facile:\n",
+        "\n",
+        "* Le operazioni di TensorFlow convertono automaticamente gli ndarrays di NumPy a tensori.\n",
+        "* Le operazioni di NumPy convertono automaticamente i tensors a ndarrays di NumPy.\rappresentazione di memoria, se possible",
+        "\n",
+        "I tensori sono convertiti esplicitamente ad ndarrays di NumPy usando il loro metodo. Queste conversioni sono tipicamente economiche dal momento che l'array e `tf.Tensor` condividono la stessa rappresentazione di memoria, se possible. Tuttavia, condividere la rappresentazione di memoria sottostante non è sempre possibile dal momento che `tf.Tensor` può essere hostata nella memoria GPU mentre gli array NumPy sono sempre supportati dalla memoria, e la conversazione coinvolge una copia dalla GPU e mentre gli array numPy sono sempre protetti dalla memoria."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 0,
+      "metadata": {
+        "colab": {},
+        "colab_type": "code",
+        "id": "lCUWzso6mbqR"
+      },
+      "outputs": [],
+      "source": [
+        "import numpy as np\n",
+        "\n",
+        "ndarray = np.ones([3, 3])\n",
+        "\n",
+        "print(\"TensorFlow operations convert numpy arrays to Tensors automatically\")\n",
+        "tensor = tf.multiply(ndarray, 42)\n",
+        "print(tensor)\n",
+        "\n",
+        "\n",
+        "print(\"And NumPy operations convert Tensors to numpy arrays automatically\")\n",
+        "print(np.add(tensor, 1))\n",
+        "\n",
+        "print(\"The .numpy() method explicitly converts a Tensor to a numpy array\")\n",
+        "print(tensor.numpy())"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "colab_type": "text",
+        "id": "PBNP8yTRfu_X"
+      },
+      "source": [
+        "## Accelerazione GPU\n",
+        "\n",
+        "Molte operazione di TensorFlow sono accelerate usando la GPU per la computazione. In assenza di annotazioni, TensorFlow decide automaticamente se usare la GPU o la CPU per un'operazione—copiando il tensore tra la memoria CPU e quella della GPU se necessario. I Tensori prodotti da un'operazione sono tipicamente supportate dalla memoria del device su cui l'operazione ha eseguito, per esempio:"
```

Comment:
operazione --> operazioni
annotazioni --> indicazioni
memoria CPU e quella --> memoria della CPU e quella
supportate dalla memoria del device su cui l'operazione ha eseguito --> supportati dalla memoria del dispositivo su cui l'operazione è stata eseguita

Reply:
Yes, all good for me